backend/controllers/walletController.js

```diff
@@ -14,9 +14,62 @@ exports.getWalletTransactions = async (req, res) => {
 
         console.log("Wallet transactions retrieved successfully.");
         res.json({ success: true, data: transactions });
-
     } catch (err) {
         console.error("Error fetching wallet transactions:", err.message);
         res.status(500).json({ success: false, message: "Server Error" });
     }
 };
+
+/**
+ * @desc    Retrieve wallet balance for the logged-in user
+ * @route   GET /getWalletBalance
+ * @return  { success: true, data: { balance: <computed_balance> } }
+ */
```

Comment:
This calculation is error prone, should we consider updating the schema of walletTransaction to have balance amount and update when there's a new transaction added